textacy/resources/concept_net.py

```diff
@@ -19,14 +37,58 @@
     "publication_url": "https://arxiv.org/abs/1612.03975",
     "description": (
         "An open, multilingual semantic network of general knowledge, "
-        "designed to help computers understand the meanings of words.",
+        "designed to help computers understand the meanings of words."
     )
 }
 DOWNLOAD_ROOT = "https://s3.amazonaws.com/conceptnet/downloads/{year}/edges/conceptnet-assertions-{version}.csv.gz"
 
 
 class ConceptNet(Resource):
     """
+    Interface to ConceptNet, a multilingual knowledge base representing common words
+    and phrases and the common-sense relationships between them.
+
+    Download the data (one time only!), and save its contents to disk::
+
+        >>> rs = ConceptNet()
+        >>> rs.download()
+        >>> rs.info
+        {'name': 'concept_net',
+         'site_url': 'http://conceptnet.io',
+         'publication_url': 'https://arxiv.org/abs/1612.03975',
+         'description': 'An open, multilingual semantic network of general knowledge, designed to help computers understand the meanings of words.'}
+
+    Access other same-language terms related to a given term in a variety of ways::
+
+        >>> rs.get_synonyms("spouse", lang="en", sense="n")
+        ['mate', 'married person', 'better half', 'partner']
+        >>> rs.get_antonyms("love", lang="en", sense="v")
+        ['detest', 'hate', 'loathe']
+        >>> rs.get_hyponyms("marriage", lang="en", sense="n")
+        ['cohabitation situation', 'union', 'legal agreement', 'ritual', 'family', 'marital status']
+
+    Note that the very first time a given relationship is accessed, the full ConceptNet db
+    must be parsed and split for fast future access. This can take a couple minutes;
+    be patient.
+
+    When passing a spaCy ``Token`` or ``Span``, the corresponding ``lang`` and ``sense``
+    are inferred automatically from the object::
+
+        >>> text = "The quick brown fox jumps over the lazy dog."
+        >>> doc = textacy.make_spacy_doc(text, lang="en")
+        >>> rs.get_synonyms(doc[1])  # quick
+        ['flying', 'fast', 'rapid', 'ready', 'straightaway', 'nimble', 'speedy', 'warm']
+        >>> rs.get_synonyms(doc[4:5])  # jumps over
```

Comment:
what happens if you are not careful, for example: `rs.get_synonyms(doc[1:3])  # quick brown`?